### DISCUSSION TITLE
Example of split config and forbid sourceOnDemand

### DISCUSSION THREAD

**Discussion by brandousov** [bluenviron/mediamtx#313] [Questions]
## Which version are you using?

v0.14.0

## Question  

1. Release notes for 0.14.1 announces `split configuration into sections` but I can not find usage example :)

2. Release notes for 0.14.1 announces `forbid usage of sourceOnDemand when source is not 'record'` but we have plans to intensively use this feature for traffic economy. Why?

3. `support relative fallback paths` (#201) - Thank you!

**Comment by brandousov** [comment]
I did short arch scheme for 2-nd question: 
https://github.com/brandousov/pix/blob/main/sourceOnDemand.svg

**Comment by dorinclisu** [comment]
I'm curious, based on your scheme since the cameras are under NAT (and no port forwarding visible), I assume you must use `source: record`, right? In this case the cameras publish themselves and sourceOnDemand does not apply.

In any case, I have the same question 2. for @aler9, because `source: rtsp://...` is a valid use case for saving traffic bandwidth with sourceOnDemand.

**Comment by dorinclisu** [comment]
Perhaps what the author meant was `forbid usage of sourceOnDemand when source is 'record'` (without "not") and then it would make perfect sense.

**Comment by aler9** [comment]
Hello, first of all, all changes can be analyzed in plain sight by looking at the history page:
https://github.com/aler9/rtsp-simple-server/commits/main

> split configuration into sections

the YML file was divided in sections by using comments
https://github.com/aler9/rtsp-simple-server/blob/main/rtsp-simple-server.yml

> forbid usage of sourceOnDemand when source is not 'record'

It's not technically possible to stop a publisher from publishing a stream or put the stream on hold until the latter is requested by someone, BUT there is a workaround.

1) start a second, local instance of rtsp-simple-server on the same server that is hosting the publisher (ffmpeg/gstreamer)

2) publish to this second, local instance

3) configure the first, main instance with

```yml
paths:
  mypath:
    source: rtsp://secondinstance/path
    sourceOnDemand: yes
```

in this way, the result is achieved - the stream will be pulled from the second instance only when it is requested.

[a comment by an automated account is not shown]
